llvm/lib/SYCLLowerIR/ESIMD/LowerESIMDSlmReservation.cpp

```diff
@@ -81,11 +82,16 @@ namespace {
 constexpr int DebugLevel = 0;
 #endif
 
+bool isTranslatesSlmInit(const Function &F) {
+  constexpr char SLM_INIT_PREFIX[] = "llvm.genx.slm.init";
+  return F.getName().starts_with(SLM_INIT_PREFIX);
+}
+
 bool isSlmInitCall(const CallInst *CI) {
   if (!CI)
     return false;
   Function *F = CI->getCalledFunction();
-  return F && esimd::isSlmInit(*F);
+  return F && (esimd::isSlmInit(*F) || isTranslatesSlmInit(*F));
```

Comment:
we shouldn't ever see pre-translation slm-init before since they should be translated in loweresimd right? if so should we instead do this?
```
  Function *F = CI->getCalledFunction();
assert(!esimd::isSlmInit(*F) && "Should have been translated already");
  return F && isTranslatesSlmInit(*F);
```